core/state/state_object.go

```diff
@@ -211,6 +211,11 @@ func (c *StateObject) Address() common.Address {
 	return c.address
 }
 
+// Sets the address of the contract/account
+func (c *StateObject) SetAddress(addr common.Address) {
+	c.address = addr
```

Comment:
I'm not familiar with this part of the code enough to properly answer this, but maybe @fjl could weigh in?

When are we using SetAddress exactly? Shouldn't we set the `dirty` flag to true? Will we ever save such a state object back into the database?